### DISCUSSION TITLE
Generic http authorization solution

### DISCUSSION THREAD

**Discussion by j3k0** [airsdk/apm#39] [Ideas]
Instead of only supporting github.com private downloads, `apm` could use a more generic approach (as first mentioned in #38).

We could allow define per-domain authorization headers in `.apm_config`:

```json
{
  "http_authorization": {
    "https://api.github.com/": "token ghp_myveryowngithubtoken",
    "https://private.domain.com/": "Basic YWxhZGRpbjpvcGVuc2VzYW1l",
  }
}
```

**Comment by marchbold** [comment]
The auth stuff is really just thrown in to get something working at the moment. 

I believe we will want to implement a proper user authentication solution at the repository, this would then add a level of authentication into the repository queries which I'd see as being added into the repository nodes in the [project file](https://github.com/airsdk/apm/wiki/ProjectDefinition). Probably more likely to be some token or something than user/pass but that's the idea.

```json
"repositories": [
        { 
            "url": "https://airnativeextensions.com/repository", 
            "credentials" : { "username": "", "password": "" }
        }
    ]
```

I see this as being essential for setting up and approving publishers as well.
